### PR TITLE
Add move to top and bottom context menu options

### PR DESCRIPTION
**Pull request by kevgrig**
**Changes**

After adding an item to a playlist, I often want to move it to the top and it's tedious to drag and drop if the playlist is large. This adds 'Move to Top' and 'Move to Bottom' options to a playlist item context menu.

**Issues**

None

### CODE REVIEW

**Comment by kevgrig** [comment]
Bump

**Comment by kevgrig** [comment]
Screenshot_20240512_215815

**Review by 1hitsong — CHANGES_REQUESTED** [review]
Testing in Waterfox in Windows 11

Display of options works as expected.

Clicking on Move to Top / Move to Bottom results in an error and items in playlist aren't moved.
image

File: src/components/shortcuts.js [hunk 1]
```diff
@@ -110,6 +110,19 @@ function showContextMenu(card, options = {}) {
...
+            if (itemsContainer) {
+                let index = 0;
+                for (const listItem of itemsContainer.querySelectorAll('.listItem')) {
+                    const playlistItemId = listItem.getAttribute('data-playlistitemid');
+                    if (playlistItemId == item.PlaylistItemId) {
+                        item.PlaylistIndex = index;
+                    }
+                    index++;
+                }
+                item.PlaylistItems = index;
```
Comment (1hitsong) [inline src/components/shortcuts.js:124]: Let's change this property name. Elsewhere in the code we already use `PlaylistItems` to mean the actual objects in a playlist. Here it's used as a count of the items in the playlist.

I suggest something like `PlaylistItemCount`.

**Review by 1hitsong — DISMISSED** [review]

File: src/components/shortcuts.js (hunk 1 above)
Comment (kevgrig) [reply src/components/shortcuts.js]: Apologies for the delay; moving houses. Pushed a new commit with the changed variable name.

**Review by thornbill — APPROVED** [review]
The context menu is getting way too complex, but fixing that will require more substantial changes. This works as expected in my testing and I don't want to block it since the menu is already a mess. :upside_down_face:

**Pull request merged**